tool/client.cc

```diff
@@ -245,19 +245,89 @@ static bool WaitForSession(SSL *ssl, int sock) {
   return true;
 }
 
+static void print_verify_details(SSL *s) {
+  long verify_err = SSL_get_verify_result(s);
+
+  if (verify_err == X509_V_OK) {
+    fprintf(stdout, "Verification: OK\n");
+  } else {
+    const char *reason = X509_verify_cert_error_string(verify_err);
+    fprintf(stdout, "Verification error: %s\n", reason);
+  }
+}
+
+static void PrintOpenSSLConnectionInfo(SSL *ssl, int show_certs) {
+  STACK_OF(X509) *sk;
+  X509 *peer;
```

Comment:
Same boolean comment for `show_certs` and we can use a unique pointer to manage `peer`. It'd also be preferable to check the return values of all the non-void function calls below as well.